src/test/java/org/mockbukkit/mockbukkit/inventory/ItemStackMirrorTest.java

```diff
@@ -0,0 +1,60 @@
+package org.mockbukkit.mockbukkit.inventory;
+
+import org.bukkit.Material;
+import org.bukkit.inventory.ItemStack;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockbukkit.mockbukkit.MockBukkitExtension;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotSame;
+
+@ExtendWith(MockBukkitExtension.class)
+class ItemStackMirrorTest
+{
+
+	@Nested
+	class Create
+	{
+
+		@Test
+		void givenValidValue_ShouldNotBeTheSame()
+		{
+			ItemStack originalItem = new ItemStackMock(Material.DIAMOND, 1);
+
+			ItemStack mirror = ItemStackMirror.create(originalItem);
+
+			assertEquals(originalItem, mirror);
+			assertNotSame(originalItem, mirror);
+		}
+
+		@Test
+		void givenAnyChangeExecutedInTheOriginalItem_ShouldBeAppliedInTheMirror()
+		{
+			ItemStack originalItem = new ItemStackMock(Material.DIAMOND, 1);
+			ItemStack mirror = ItemStackMirror.create(originalItem);
+
+			originalItem.setAmount(5);
+
+			assertEquals(5, originalItem.getAmount());
+			assertEquals(originalItem, mirror);
+			assertNotSame(originalItem, mirror);
+		}
+
+		@Test
+		void givenAnyChangeExecutedInTheMirrorItem_ShouldBeAppliedInTheOriginal()
+		{
+			ItemStack originalItem = new ItemStackMock(Material.DIAMOND, 1);
+			ItemStack mirror = ItemStackMirror.create(originalItem);
+
+			mirror.setAmount(5);
+
+			assertEquals(5, mirror.getAmount());
+			assertEquals(originalItem, mirror);
```

Comment:
Would this also pass the test?
```java
assertEquals(mirror, originalItem);
```
I looked into assert equals code, and it seems like it only checks if `obj1.equals(obj2)` not `obj2.equals(obj1)`

In this case, it might cause some issues, ItemStackMirror#equals needs to be tested as well

Reply:
Check, this is ok.